src/services/utilities.ts

```diff
@@ -790,6 +790,8 @@ namespace ts {
     }
 
     export function getContextualTypeFromParentOrAncestorTypeNode(node: Expression, checker: TypeChecker): Type | undefined {
+        if (node.flags & (NodeFlags.JSDoc & ~NodeFlags.JavaScriptFile)) return undefined;

```

Comment:
Should this have been something like the following?

```ts
if (node.flags & NodeFlags.JSDoc && !(node.flags & NodeFlags.JavaScriptFile)) return undefined;
```

Reply:
@DanielRosenwasser Yes. Do I need to change it?

Reply:
If you wouldn't mind. I think the regression test will look like this

```js
/**
 * @type {"foo" | "bar"}
 */
let x = "foo"$
```

with a find-all-references on `$` resulting in both the expression and the string literal type `"foo"`.